packages/model-viewer/src/features/scene-graph/material.ts

```diff
@@ -64,31 +114,61 @@ export class Material extends ThreeDOMElement implements MaterialInterface {
 
       if (normalTexture != null && normalMap != null) {
         normalTextures.add(normalMap);
+      } else {
+        normalTexture = {index: -1};
       }
 
       if (occlusionTexture != null && aoMap != null) {
         occlusionTextures.add(aoMap);
+      } else {
+        occlusionTexture = {index: -1};
       }
 
       if (emissiveTexture != null && emissiveMap != null) {
         emissiveTextures.add(emissiveMap);
+      } else {
+        emissiveTexture = {index: -1};
       }
     }
 
-    if (normalTextures.size > 0) {
-      this[$normalTexture] =
-          new TextureInfo(onUpdate, gltf, normalTexture!, normalTextures);
-    }
-
-    if (occlusionTextures.size > 0) {
-      this[$occlusionTexture] =
-          new TextureInfo(onUpdate, gltf, occlusionTexture!, occlusionTextures);
-    }
-
-    if (emissiveTextures.size > 0) {
-      this[$emissiveTexture] =
-          new TextureInfo(onUpdate, gltf, emissiveTexture!, emissiveTextures);
-    }
+    this[$normalTexture] = new TextureInfo(
+        onUpdate,
+        gltf,
+        normalTexture!,
+        normalTextures,
+        // Applicator provides method for applying a texture to a material.
+        TextureApplicator[$provideApplicator](
+            onUpdate,
+            correlatedMaterials,
+            TextureUsage.Normal,
+            LinearEncoding,
+            RGBFormat));
```

Comment:
Considering the encoding default is directly related to the texture usage, which is already specified here, how about we remove this input and handle the logic directly inside the applicator? Also, the occlusion and MR textures are usually the same (occlusion simply uses the red channel). I believe you have to set the format based on the incoming buffer, which I think the loader already does.

Reply:
I can move the encoding logic, no problem.
I see for the occlusion it's set to RGB after loading, I guess that's to be expected if it's sharing with the MR map.

Reply:
Resolved